.github/workflows/codeql-analysis.yml

```diff
@@ -16,10 +16,10 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
`v4` would also work; however, my understanding is that specifying the full version and SHA is preferable in general, because it is more difficult for a malicious hacker to compromise your code via a compromised version of the GitHub action. See https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Technically, I suppose that since this is an official GitHub action, it doesn't fall under the category of "third-party actions." I've just gotten into the habit of using the full SHA wherever I can. Then I let Dependabot, Renovate Bot, or similar tools suggest updates to my GH Actions when there are updates available.

Let me know if you prefer simply `v4` or `v4.2.2`, and I can update my PR accordingly.

Thanks!

Reply:
Let's just go with the version number, since it's an official GitHub action.

Would "v4" end up using 4.2.2, since it is the latest version of the "4.0" release? I'd like to keep it as simple as possible.